nat-lab/tests/test_dns.py

```diff
@@ -843,3 +844,64 @@ async def test_dns_wildcarded_records() -> None:
         await query_dns(connection_beta, "myserviceC.beta.nord", beta.ip_addresses)
         await query_dns(connection_beta, "myserviceD.yoko.nord", beta.ip_addresses)
         await query_dns(connection_beta, "hisservice.johnny.nord", alpha.ip_addresses)
+
+
+@pytest.mark.asyncio
+async def test_dns_ttl_value() -> None:
+    async with AsyncExitStack() as exit_stack:
+        FIRST_DNS_SERVER = "10.0.80.83"
+        SECOND_DNS_SERVER = "10.0.80.82"
+
+        EXPECTED_TTL_VALUE = 1234
+
+        api, (_, _) = setup_api([(False, IPStack.IPv4v6), (False, IPStack.IPv4v6)])
+        env = await setup_mesh_nodes(
+            exit_stack,
+            [
+                SetupParameters(
+                    connection_tag=ConnectionTag.DOCKER_CONE_CLIENT_1,
+                    connection_tracker_config=generate_connection_tracker_config(
+                        ConnectionTag.DOCKER_CONE_CLIENT_1,
+                        derp_1_limits=ConnectionLimits(1, 1),
+                    ),
+                    features=TelioFeatures(
+                        dns=Dns(exit_dns=None, ttl_value=EXPECTED_TTL_VALUE)
+                    ),
+                ),
+                SetupParameters(
+                    connection_tag=ConnectionTag.DOCKER_CONE_CLIENT_2,
+                    connection_tracker_config=generate_connection_tracker_config(
+                        ConnectionTag.DOCKER_CONE_CLIENT_2,
+                        derp_1_limits=ConnectionLimits(1, 1),
+                    ),
+                ),
+            ],
+            provided_api=api,
+        )
+        client_alpha, *_ = env.clients
+        connection_alpha, *_ = [conn.connection for conn in env.connections]
+
+        await client_alpha.enable_magic_dns([FIRST_DNS_SERVER, SECOND_DNS_SERVER])
+        await asyncio.sleep(1)
```

Comment:
Will this not become a source of flakiness in the future?

Reply:
I agree with @tomaszklak as this sleep is pretty much guaranteed to be flaky. We need some better mechanism to wait for DNS server to become ready, which would mean to wait until WG session to internal virtual host is established.

In practical terms currently we do not report events for "internal" peers (like WG-STUN or internal virtual host acting as DNS). Therefore I see two option -> one is coupling to logs, although I am not 100% sure it is the best option, and the other is conditianlly reporting internal nodes for tests (e.g. enabled via feature config).